nikola/post.py

```diff
@@ -153,15 +153,17 @@ def __init__(
         self._dependency_uptodate_page = defaultdict(list)
         self._depfile = defaultdict(list)
 
+        # Load internationalized metadata
+        for lang in self.translations:
+            if os.path.isfile(get_translation_candidate(self.config, self.source_path, lang)):
+                self.translated_to.add(lang)
+
         default_metadata, self.newstylemeta = get_meta(self, self.config['FILE_METADATA_REGEXP'], self.config['UNSLUGIFY_TITLES'])
 
         self.meta = Functionary(lambda: None, self.default_lang)
         self.meta[self.default_lang] = default_metadata
 
-        # Load internationalized metadata
         for lang in self.translations:
-            if os.path.isfile(get_translation_candidate(self.config, self.source_path, lang)):
-                self.translated_to.add(lang)
```

Comment:
Perhaps we should add a fallback to `get_translation_candiate` if `self.translated_to` is empty? There also seems to be a `lang != self.default_lang` case that will return the source path (shouldn’t it be flipped?)